codegen/src/generator/mod.rs

```diff
@@ -88,10 +89,10 @@ fn generate_client<P>(service: &Service, protocol_generator: &P) -> String
         }}
 
         impl<P> {type_name}<P, Client> where P: ProvideAwsCredentials {{
-            pub fn new(credentials_provider: P, region: region::Region) -> Self {{
-                let mut client = Client::new();
-                client.set_redirect_policy(RedirectPolicy::FollowNone);
-               {type_name}::with_request_dispatcher(client, credentials_provider, region)
+            pub fn new(credentials_provider: P, region: region::Region) -> Result<Self, String> {{
```

Comment:
Instead of a `String` type for the error component of the `Result` here we should use something that implements `std::error::Error` to make error composition easier on people who use our crate.